src/net_processing.cpp

```diff
@@ -4054,15 +4067,19 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             }
 
             if (pindex->nHeight >= m_chainman.ActiveChain().Height() - MAX_BLOCKTXN_DEPTH) {
-                CBlock block;
-                const bool ret{m_chainman.m_blockman.ReadBlockFromDisk(block, *pindex)};
-                assert(ret);
-
-                SendBlockTransactions(pfrom, *peer, block, req);
-                return;
+                block_pos = pindex->GetBlockPos();
             }
         }
 
+        if (!block_pos.IsNull()) {
+            CBlock block;
+            const bool ret{m_chainman.m_blockman.ReadBlockFromDisk(block, block_pos)};
+            assert(ret);
```

Comment:
Worth adding a comment here, it's not immediate that this assertion would never fail because `MAX_BLOCKTXN_DEPTH < MIN_BLOCKS_TO_KEEP`.

Reply:
Rebased and added a comment to address this.

Reply:
Would it be woth having a static assert for this when defining `MAX_BLOCKTXN_DEPTH`?

```
static_assert(MAX_BLOCKTXN_DEPTH < MIN_BLOCKS_TO_KEEP, "MIN_BLOCKS_TO_KEEP too low");
```

This was not obvious to me either, but I'm not too familiar with this part of net_processing

Reply:
Would it be worth having a static assert for this when defining `MAX_BLOCKTXN_DEPTH`?

```
static_assert(MAX_BLOCKTXN_DEPTH < MIN_BLOCKS_TO_KEEP, "MIN_BLOCKS_TO_KEEP too low");
```

This was not obvious to me either, but I'm not too familiar with this part of net_processing